src/features/class-query-block-context.php

```diff
@@ -48,60 +49,81 @@ public function __construct(
 	 * Boot the feature.
 	 */
 	public function boot(): void {
-		add_filter( 'render_block_context', [ $this, 'filter_query_context' ], 10, 2 );
+		add_filter( 'render_block_context', [ $this, 'filter_query_context' ], 10, 3 );
 	}
 
 	/**
 	 * Filters the context provided to a query block.
 	 *
 	 * @param array<string, mixed>                 $context      Default context.
 	 * @param array{"attrs": array<string, mixed>} $parsed_block Block being rendered.
+	 * @param WP_Block|null                        $parent_block Parent block, if any.
 	 * @return array<string, mixed> Updated context.
 	 */
-	public function filter_query_context( $context, $parsed_block ) {
-		$current_block     = new Parsed_Block( $parsed_block );
-		$plugin_block_type = $this->block_type_registry->get_registered( 'wp-curate/query' );
+	public function filter_query_context( $context, $parsed_block, $parent_block ) {
+		$current_block      = new Parsed_Block( $parsed_block );
+		$current_block_type = $this->block_type_registry->get_registered( (string) $current_block->block_name() );
+		$plugin_block_type  = $this->block_type_registry->get_registered( 'wp-curate/query' );
 
-		if ( ! $plugin_block_type instanceof WP_Block_Type || $current_block->block_name() !== $plugin_block_type->name ) {
-			return $context;
-		}
-
-		$curated_posts = new Recorded_Curated_Posts(
-			// Deduplication is achieved by both recording to and excluding from the same place.
-			history: $this->history,
-			origin: new Must_Include_Curated_Posts(
-				qv: $this->stop_queries_var,
-				origin: new Plugin_Curated_Posts(
-					queries: new Variable_Post_Queries(
-						input: function () use ( $parsed_block ) {
-							$main_query = $this->main_query->query_object();
+		// Provide 'query' context to the plugin's query block.
+		if ( $plugin_block_type instanceof WP_Block_Type && $current_block->block_name() === $plugin_block_type->name ) {
+			$curated_posts = new Recorded_Curated_Posts(
+				// Deduplication is achieved by both recording to and excluding from the same place.
+				history: $this->history,
```

Comment:
First time I've seen named arguments. I had to look at the docs to see what this was all about! TIL.

Reply:
Yeah, I like them. With that and [constructor property promotion](https://github.com/alleyinteractive/wp-curate/blob/b0d82733d8c0aebb698c16ff434c90ac742db71b/src/class-recorded-curated-posts.php#L17-L26) (which I was skeptical of at first), you can make object initialization pretty clean and code-free.